storm-core/src/jvm/backtype/storm/blobstore/NimbusBlobStore.java

```diff
@@ -0,0 +1,408 @@
+        return client.getClient().getBlobMeta(key);
+      }
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  protected void setBlobMetaToExtend(String key, SettableBlobMeta meta)
+      throws AuthorizationException, KeyNotFoundException {
+    try {
+      synchronized(client) {
+        client.getClient().setBlobMeta(key, meta);
+      }
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void deleteBlob(String key) throws AuthorizationException, KeyNotFoundException {
+    try {
+      synchronized(client) {
+        client.getClient().deleteBlob(key);
+      }
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void createStateInZookeeper(String key) {
+    try {
+      synchronized(client) {
+        client.getClient().createStateInZookeeper(key);
+      }
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public InputStreamWithMeta getBlob(String key) throws AuthorizationException, KeyNotFoundException {
+    try {
+      synchronized(client) {
+        return new NimbusDownloadInputStream(client.getClient().beginBlobDownload(key));
+      }
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<String> listKeys() {
+    try {
+      synchronized(client) {
+        return new NimbusKeyIterator(client.getClient().listBlobs(""));
+      }
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int getBlobReplication(String key) throws AuthorizationException, KeyNotFoundException {
+    try {
+      return client.getClient().getBlobReplication(key);
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int updateBlobReplication(String key, int replication) throws AuthorizationException, KeyNotFoundException {
+    try {
+      return client.getClient().updateBlobReplication(key, replication);
+    } catch (AuthorizationException | KeyNotFoundException exp) {
+      throw exp;
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public boolean setClient(Map conf, NimbusClient client) {
+    this.client = client;
+    if (conf != null) {
+      this.bufferSize = Utils.getInt(conf.get(Config.STORM_BLOBSTORE_INPUTSTREAM_BUFFER_SIZE_BYTES), bufferSize);
+    }
+    return true;
+  }
+
+  @Override
+  protected void finalize() {
```

Comment:
It relieves only a client object before gc, might not be a big issue but no harm in keeping it I think.